### ISSUE TITLE
Upgrde transformers version

### ISSUE THREAD

**Issue by hearwish** [expert-kit/expert-kit#123]
## Summary

Expert-Kit currently uses an older `transformers` baseline in the Python integration environment. This limits support for newer model architectures and makes recently added upstream model implementations harder to use or validate.

During Qwen3-30B-A3B deployment validation, the Expert-Kit Python integration also failed with a `Tensor + tuple` crash in the Qwen3 MoE path. In issue #122, the reproduced environment uses:

```text
transformers==5.0.0
torch==2.7.0+cu126
Python==3.12.13
```

 This is reasonable because Expert-Kit's Qwen3 integration monkey-patches `transformers.models.qwen3_moe`, so it is sensitive to upstream Qwen3 MoE implementation details.

This issue proposes upgrading the tested and documented `transformers` version for Expert-Kit Python integrations, both to support newer models and to make Qwen3 MoE compatibility easier to validate. Because some integration paths depend on internal `transformers` model APIs, the upgraded dependency should still use a bounded compatible version range.
## Recommendation

Use the following dependency range for Expert-Kit Python model integrations:

```text
transformers>=5.10.4,<5.13
```

Use the following version as the primary deployment validation target:

```text
transformers==5.12.1
```

Also test the lower bound:

```text
transformers==5.10.4
```

### Why upgrade from `5.0.0`

The currently reproduced environment uses `transformers==5.0.0`, which is an early v5 release. Since then, the v5 modeling stack has continued to evolve, including fixes and additions for newer architectures, generation behavior, tokenizer behavior, and model-specific implementations.

Using a later stable patch release gives Expert-Kit a better baseline for:

- newer upstream model architecture support
- model-specific implementation fixes
- generation stack fixes
- tokenizer and processor compatibility
- MoE-related modeling improvements
- better alignment with downstream inference stacks

### Why choose `5.12.1` as the primary validation version

`5.12.1` is a recent stable patch release and is a better single-version validation target than `5.0.0`.

It gives Expert-Kit access to newer upstream model and generation fixes while avoiding the instability of installing directly from `main`.

Recommended validation command:

```bash
uv pip install "transformers==5.12.1"
```

### Why use a bounded range instead of latest-only

Some Expert-Kit Python integration paths rely on `transformers` internals or monkey-patch upstream model implementations. . This makes Expert-Kit vulnerable to upstream API changes in areas such as:

- model block `forward()` return contracts
- decoder layer expectations
- router logits handling
- model output structures

For this reason, the dependency should not be unbounded, for example:

```text
transformers>=5.12.1
```

An unbounded range may silently accept future versions with breaking internal changes.

The proposed range:

```text
transformers>=5.10.4,<5.13
```

keeps the dependency modern enough for currently validated newer models, while limiting exposure to future minor-version behavior changes until they are tested.
It also creates a clearer compatibility window for validating additional newer models.

### 1. Update Python dependency constraints

Update the relevant Python dependency file, for example `pyproject.toml` or the integration package dependency list:

```toml
transformers = ">=5.10.4,<5.13"
```

If the project prefers exact reproducibility for demos, document the tested version:

```text
transformers==5.12.1
```

### 2. Document the model support baseline

Update Expert-Kit Python integration docs to state the tested dependency baseline for model support:

```text
Python: 3.12.x
torch: 2.7.0+cu126
transformers: 5.12.1
expertkit_transport: 0.1.0
```

The docs should explain that newer model support depends on the `transformers` version, and that users should use the documented baseline when validating recently added model architectures.

### 3. Add compatibility smoke tests

Add smoke tests for supported model integration paths against the supported `transformers` versions.

For the Qwen3 regression case, the test should verify:

- the intercepted MoE block returns the type expected by upstream `transformers`
- generation does not fail with `Tensor + tuple`
- router logits handling does not alter the upstream return contract

For newer model support in general, smoke tests should verify:

- the model class can be imported from `transformers`
- tokenizer/model config loading works
- a minimal forward or generation path works with Expert-Kit integration enabled, where applicable

Suggested test matrix:
```text
transformers==5.10.4
transformers==5.12.1
```
